packages/@o3r/eslint-plugin/src/rules/o3r-widget-tags/o3r-widget-tags.ts

```diff
@@ -0,0 +1,168 @@
+import { createRule } from '../utils';
+
+const o3rWidgetParameterPattern = '^[a-zA-Z0-9-_:.]+$';
+
+export interface O3rWidgetTagsRuleOption {
+  [widgetName: string]: {
+    [paramName: string]: 'string' | 'number' | 'boolean' | 'string[]' | 'number[]' | 'boolean[]';
```

Comment:
Could be externalized and apply to the `enum` above to make sure they are aligned